Debian currently ships with lmodern 1.010x, while context expects the 
font names from the newer release(s) of lmodern.
Patch type-otf.tex to work around this problem.
Patch from Mojca Miklavec
---
 tex/context/base/type-otf.tex |  117 +++++++++++++++++++++---------------------
 1 file changed, 60 insertions(+), 57 deletions(-)

Index: context-2008.05.21/tex/context/base/type-otf.tex
===================================================================
--- context-2008.05.21.orig/tex/context/base/type-otf.tex	2008-05-22 00:51:27.000000000 +0200
+++ context-2008.05.21/tex/context/base/type-otf.tex	2008-05-22 00:51:44.000000000 +0200
@@ -11,6 +11,8 @@
 %C therefore copyrighted by \PRAGMA. See mreadme.pdf for
 %C details.
 
+%D This file has been modified for Debian only to be able to use it with old LM fonts.
+
 %D Actually, \LUATEX\ does not need all these definitions since it can
 %D consult its database. However, here we use the names that we used
 %D in good old \TEX, and these may differ a bit. Here we also see
@@ -125,69 +127,70 @@
     \definefontsynonym [LMRoman10-Italic]      [file:lmroman10-italic]      [features=default]
     \definefontsynonym [LMRoman12-Italic]      [file:lmroman12-italic]      [features=default]
     \definefontsynonym [LMRoman10-BoldItalic]  [file:lmroman10-bolditalic]  [features=default]
-    \definefontsynonym [LMRoman8-Oblique]      [file:lmromanslant8-regular] [features=default]
-    \definefontsynonym [LMRoman9-Oblique]      [file:lmromanslant9-regular] [features=default]
-    \definefontsynonym [LMRoman10-Oblique]     [file:lmromanslant10-regular][features=default]
-    \definefontsynonym [LMRoman12-Oblique]     [file:lmromanslant12-regular][features=default]
-    \definefontsynonym [LMRoman17-Oblique]     [file:lmromanslant17-regular][features=default]
-    \definefontsynonym [LMRoman10-BoldOblique] [file:lmromanslant10-bold]   [features=default]
-    \definefontsynonym [LMRoman10-Demi]        [file:lmromandemi10-regular] [features=default]
-    \definefontsynonym [LMRoman10-DemiOblique] [file:lmromandemi10-oblique] [features=default]
-    \definefontsynonym [LMRoman10-CapsRegular] [file:lmromancaps10-regular] [features=default] % features=smallcaps?
-    \definefontsynonym [LMRoman10-CapsOblique] [file:lmromancaps10-oblique] [features=default]
-
-    \definefontsynonym [LMRoman10-Dunhill]        [file:lmromandunh10-regular] [features=default]
-    \definefontsynonym [LMRoman10-DunhillOblique] [file:lmromandunh10-oblique] [features=default]
-    \definefontsynonym [LMRoman10-Unslanted]      [file:lmromanunsl10-regular] [features=default]
+    \definefontsynonym [LMRoman8-Oblique]      [file:lmroman8-oblique]      [features=default]
+    \definefontsynonym [LMRoman9-Oblique]      [file:lmroman9-oblique]      [features=default]
+    \definefontsynonym [LMRoman10-Oblique]     [file:lmroman10-oblique]     [features=default]
+    \definefontsynonym [LMRoman12-Oblique]     [file:lmroman12-oblique]     [features=default]
+    \definefontsynonym [LMRoman17-Oblique]     [file:lmroman17-oblique]     [features=default]
+    \definefontsynonym [LMRoman10-BoldOblique] [file:lmroman10-boldoblique] [features=default]
+    \definefontsynonym [LMRoman10-Demi]        [file:lmroman10-demi]        [features=default]
+    \definefontsynonym [LMRoman10-DemiOblique] [file:lmroman10-demioblique] [features=default]
+    \definefontsynonym [LMRoman10-CapsRegular] [file:lmroman10-capsregular] [features=default] % features=smallcaps?
+    \definefontsynonym [LMRoman10-CapsOblique] [file:lmroman10-capsoblique] [features=default]
+
+    \definefontsynonym [LMRoman10-Dunhill]        [file:lmroman10-dunhill]        [features=default]
+    \definefontsynonym [LMRoman10-DunhillOblique] [file:lmroman10-dunhilloblique] [features=default]
 \stoptypescript
 
 \starttypescript [sans] [modern,latin-modern]
-    \definefontsynonym [LMSans8-Regular]               [file:lmsans8-regular]          [features=default]
-    \definefontsynonym [LMSans9-Regular]               [file:lmsans9-regular]          [features=default]
-    \definefontsynonym [LMSans10-Regular]              [file:lmsans10-regular]         [features=default]
-    \definefontsynonym [LMSans12-Regular]              [file:lmsans12-regular]         [features=default]
-    \definefontsynonym [LMSans17-Regular]              [file:lmsans17-regular]         [features=default]
-    \definefontsynonym [LMSans10-Bold]                 [file:lmsans10-bold]            [features=default]
-    \definefontsynonym [LMSans8-Oblique]               [file:lmsans8-oblique]          [features=default]
-    \definefontsynonym [LMSans9-Oblique]               [file:lmsans9-oblique]          [features=default]
-    \definefontsynonym [LMSans10-Oblique]              [file:lmsans10-oblique]         [features=default]
-    \definefontsynonym [LMSans12-Oblique]              [file:lmsans12-oblique]         [features=default]
-    \definefontsynonym [LMSans17-Oblique]              [file:lmsans17-oblique]         [features=default]
-    \definefontsynonym [LMSans10-BoldOblique]          [file:lmsans10-boldoblique]     [features=default]
-
-    \definefontsynonym [LMSans10-DemiCondensed]        [file:lmsansdemicond10-regular] [features=default]
-    \definefontsynonym [LMSans10-DemiCondensedOblique] [file:lmsansdemicond10-oblique] [features=default]
-
-    \definefontsynonym [LMSansQuotation8-Regular]      [file:lmsansquot8-regular]      [features=default]
-    \definefontsynonym [LMSansQuotation8-Bold]         [file:lmsansquot8-bold]         [features=default]
-    \definefontsynonym [LMSansQuotation8-Oblique]      [file:lmsansquot8-oblique]      [features=default]
-    \definefontsynonym [LMSansQuotation8-BoldOblique]  [file:lmsansquot8-boldoblique]  [features=default]
+    \definefontsynonym [LMSans8-Regular]               [file:lmsans8-regular]               [features=default]
+    \definefontsynonym [LMSans9-Regular]               [file:lmsans9-regular]               [features=default]
+    \definefontsynonym [LMSans10-Regular]              [file:lmsans10-regular]              [features=default]
+    \definefontsynonym [LMSans12-Regular]              [file:lmsans12-regular]              [features=default]
+    \definefontsynonym [LMSans17-Regular]              [file:lmsans17-regular]              [features=default]
+    \definefontsynonym [LMSans10-Bold]                 [file:lmsans10-bold]                 [features=default]
+    \definefontsynonym [LMSans8-Oblique]               [file:lmsans8-oblique]               [features=default]
+    \definefontsynonym [LMSans9-Oblique]               [file:lmsans9-oblique]               [features=default]
+    \definefontsynonym [LMSans10-Oblique]              [file:lmsans10-oblique]              [features=default]
+    \definefontsynonym [LMSans12-Oblique]              [file:lmsans12-oblique]              [features=default]
+    \definefontsynonym [LMSans17-Oblique]              [file:lmsans17-oblique]              [features=default]
+    \definefontsynonym [LMSans10-BoldOblique]          [file:lmsans10-boldoblique]          [features=default]
+
+    \definefontsynonym [LMSans10-DemiCondensed]        [file:lmsans10-demicondensed]        [features=default]
+    \definefontsynonym [LMSans10-DemiCondensedOblique] [file:lmsans10-demicondensedoblique] [features=default]
+
+    \definefontsynonym [LMSansQuotation8-Regular]      [file:lmsansquotation8-regular]      [features=default]
+    \definefontsynonym [LMSansQuotation8-Bold]         [file:lmsansquotation8-bold]         [features=default]
+    \definefontsynonym [LMSansQuotation8-Oblique]      [file:lmsansquotation8-oblique]      [features=default]
+    \definefontsynonym [LMSansQuotation8-BoldOblique]  [file:lmsansquotation8-boldoblique]  [features=default]
 \stoptypescript
 
 \starttypescript [mono] [modern,latin-modern,modern-vari,latin-modern-vari,modern-cond,latin-modern-cond]
-    \definefontsynonym [LMTypewriter8-Regular]                [file:lmmono8-regular]
-    \definefontsynonym [LMTypewriter9-Regular]                [file:lmmono9-regular]
-    \definefontsynonym [LMTypewriter10-Regular]               [file:lmmono10-regular]
-    \definefontsynonym [LMTypewriter12-Regular]               [file:lmmono12-regular]
-    \definefontsynonym [LMTypewriter10-Italic]                [file:lmmono10-italic]
-    \definefontsynonym [LMTypewriter10-Oblique]               [file:lmmonoslant10-regular]
-    \definefontsynonym [LMTypewriter10-CapsRegular]           [file:lmmonocaps10-regular] % features=smallcaps?
-    \definefontsynonym [LMTypewriter10-CapsOblique]           [file:lmmonocaps10-oblique]
-
-    \definefontsynonym [LMTypewriter10-Light]                 [file:lmmonolt10-regular]
-    \definefontsynonym [LMTypewriter10-LightOblique]          [file:lmmonolt10-oblique]
-    \definefontsynonym [LMTypewriter10-LightCondensed]        [file:lmmonoltcond10-regular]
-    \definefontsynonym [LMTypewriter10-LightCondensedOblique] [file:lmmonoltcond10-oblique]
-
-    \definefontsynonym [LMTypewriter10-Dark]                  [file:lmmonolt10-bold]
-    \definefontsynonym [LMTypewriter10-DarkOblique]           [file:lmmonolt10-boldoblique]
-
-    \definefontsynonym [LMTypewriterVarWd10-Regular]          [file:lmmonoproplt10-regular]     [features=default]
-    \definefontsynonym [LMTypewriterVarWd10-Oblique]          [file:lmmonoproplt10-oblique]     [features=default]
-    \definefontsynonym [LMTypewriterVarWd10-Light]            [file:lmmonoprop10-regular]       [features=default]
-    \definefontsynonym [LMTypewriterVarWd10-LightOblique]     [file:lmmonoprop10-oblique]       [features=default]
-    \definefontsynonym [LMTypewriterVarWd10-Dark]             [file:lmmonoproplt10-bold]        [features=default]
-    \definefontsynonym [LMTypewriterVarWd10-DarkOblique]      [file:lmmonoproplt10-boldoblique] [features=default]
+    \definefontsynonym [LMTypewriter8-Regular]                [file:lmtypewriter8-regular]
+    \definefontsynonym [LMTypewriter9-Regular]                [file:lmtypewriter9-regular]
+    \definefontsynonym [LMTypewriter10-Regular]               [file:lmtypewriter10-regular]
+    \definefontsynonym [LMTypewriter12-Regular]               [file:lmtypewriter12-regular]
+    \definefontsynonym [LMTypewriter10-Italic]                [file:lmtypewriter10-italic]
+    \definefontsynonym [LMTypewriter10-Oblique]               [file:lmtypewriter10-oblique]
+    \definefontsynonym [LMTypewriter10-CapsRegular]           [file:lmtypewriter10-capsregular] % features=smallcaps?
+    \definefontsynonym [LMTypewriter10-CapsOblique]           [file:lmtypewriter10-capsoblique]
+
+    \definefontsynonym [LMTypewriter10-Light]                 [file:lmtypewriter10-light]
+    \definefontsynonym [LMTypewriter10-LightOblique]          [file:lmtypewriter10-lightoblique]
+    \definefontsynonym [LMTypewriter10-LightCondensed]        [file:lmtypewriter10-lightcondensed]
+    \definefontsynonym [LMTypewriter10-LightCondensedOblique] [file:lmtypewriter10-lightcondensedoblique]
+
+    \definefontsynonym [LMTypewriter10-Dark]                  [file:lmtypewriter10-dark]
+    \definefontsynonym [LMTypewriter10-DarkOblique]           [file:lmtypewriter10-darkoblique]
+
+    \definefontsynonym [LMTypewriter10-Unslanted]             [file:lmtypewriter10-unslanted]
+
+    \definefontsynonym [LMTypewriterVarWd10-Regular]          [file:lmtypewritervarwd10-regular]          [features=default]
+    \definefontsynonym [LMTypewriterVarWd10-Oblique]          [file:lmtypewritervarwd10-oblique]          [features=default]
+    \definefontsynonym [LMTypewriterVarWd10-Light]            [file:lmtypewritervarwd10-light]            [features=default]
+    \definefontsynonym [LMTypewriterVarWd10-LightOblique]     [file:lmtypewritervarwd10-lightoblique]     [features=default]
+    \definefontsynonym [LMTypewriterVarWd10-Dark]             [file:lmtypewritervarwd10-dark]             [features=default]
+    \definefontsynonym [LMTypewriterVarWd10-DarkOblique]      [file:lmtypewritervarwd10-darkoblique]      [features=default]
 \stoptypescript
 
 \starttypescript [math] [modern,latin-modern]
